File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    day = 0  # 배포까지 걸리는 시간
    count = 0  # 그날 배포되는 기능의 수
    
    while progresses:
        # 첫 번째 기능이 100% 이상이 될 때까지 일수 증가
        if (progresses[0] + day * speeds[0]) >= 100:
            progresses.pop(0)
            speeds.pop(0)
            count += 1
        else:
            if count > 0:  # 배포할 기능이 있다면
                answer.append(count)
                count = 0  # 배포된 후에는 count 초기화
            day += 1  # 첫 번째 기능이 100%에 도달하지 않았다면 다음 날로 넘어감
            
    if count > 0:  # 마지막으로 남은 기능 배포
        answer.append(count)
    
    return answer

# 함수 실행 예
# print(solution([93, 30, 55], [1, 30, 5]))

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    
    let day = 0; // 일 카운트
    let count = 0; // 기능 완료 수 카운트
    
    while(progresses.length !== 0) { // 모든 작업이 끝날 때까지
        if(100 <= progresses[0] + speeds[0] * day) { // 첫번째 작업 완료면
            progresses.shift(); // 작업목록에서 제거
            speeds.shift(); // 작업 속도도 같이 제거
            count ++; // 기능 완료이므로 +1 증가
        } else {
            if(count>0) answer.push(count)

            day ++ // 작업이 100 미만이라면 다음날로
            count = 0; // count 다시 세줘야하니깐 0으로 초기롸
        }
    }
    if(count>0) answer.push(count)


    return answer;
}